src/pages/resultsView/ResultsViewURLWrapper.ts

```diff
@@ -117,87 +117,151 @@ export type ResultsViewURLQuery = {
+        oncoprint_sortby: { isSessionProp: false },
+        oncoprint_cluster_profile: { isSessionProp: false },
+        oncoprint_sort_by_mutation_type: {
+            isSessionProp: false,
+        },
+        oncoprint_sort_by_drivers: { isSessionProp: false },
+        generic_assay_groups: { isSessionProp: false },
+        exclude_germline_mutations: { isSessionProp: false },
+        hide_unprofiled_samples: { isSessionProp: false },
+        patient_enrichments: { isSessionProp: false },
+
+        comparison_subtab: { isSessionProp: false },
+        comparison_overlapStrategy: { isSessionProp: false },
+        comparison_selectedGroups: { isSessionProp: false },
+        comparison_groupOrder: { isSessionProp: false },
+        comparison_selectedEnrichmentEventTypes: {
+            isSessionProp: true,
+        },
+
+        // plots
+        plots_horz_selection: {
+            isSessionProp: false,
+            nestedObjectProps: PlotsSelectionParamProps,
+        },
+        plots_vert_selection: {
+            isSessionProp: false,
+            nestedObjectProps: PlotsSelectionParamProps,
+        },
+        plots_coloring_selection: {
+            isSessionProp: false,
+            nestedObjectProps: PlotsColoringParamProps,
+        },
+
+        // mutations
+        mutations_gene: {
+            isSessionProp: false,
+        },
+        mutations_transcript_id: {
+            isSessionProp: false,
+        },
+
+        // session props here
+        gene_list: {
+            isSessionProp: true,
+            doubleURIEncode: true,
+        },
+        cancer_study_list: {
+            isSessionProp: true,
+            aliases: ['cancer_study_id'],
+        },
+        case_ids: { isSessionProp: true },
+        sample_list_ids: { isSessionProp: true },
+        case_set_id: { isSessionProp: true },
+        profileFilter: {
+            isSessionProp: true,
+            aliases: ['data_priority'],
+        },
+        RPPA_SCORE_THRESHOLD: { isSessionProp: true },
+        Z_SCORE_THRESHOLD: { isSessionProp: true },
+        geneset_list: { isSessionProp: true },
+        genetic_profile_ids_PROFILE_MUTATION_EXTENDED: {
+            isSessionProp: true,
+        },
+        genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION: {
+            isSessionProp: true,
+        },
+        genetic_profile_ids_PROFILE_MRNA_EXPRESSION: {
+            isSessionProp: true,
+        },
+        genetic_profile_ids_PROFILE_PROTEIN_EXPRESSION: {
+            isSessionProp: true,
+        },
+        genetic_profile_ids_PROFILE_GENESET_SCORE: {
+            isSessionProp: true,
+        },
+        genetic_profile_ids_GENERIC_ASSAY: {
+            isSessionProp: true,
+        },
+        genetic_profile_ids: { isSessionProp: true },
+        comparison_createdGroupsSessionId: {
+            isSessionProp: true,
+        },
+    } as PropertiesMap<ResultsViewURLQuery>,
+    (propertySpec, propertyName) => {
```

Comment:
curious why we do this merge instead of just adding property to the objects listed in propertiesMap?